tests/integration/test_utils/data_source_config/base.py

```diff
@@ -17,9 +17,13 @@
     from pytest import FixtureRequest
 
 
+_ColumnTypes = TypeVar("_ColumnTypes")
+
+
 @dataclass(frozen=True)
-class DataSourceTestConfig(ABC):
+class DataSourceTestConfig(ABC, Generic[_ColumnTypes]):
     name: Optional[str] = None
+    column_types: Union[Dict[str, _ColumnTypes], None] = None
```

Comment:
I think it would apply to spark as well, right? At least data frames, not sure if there would be any difference writing it out to disk for a file backed system.